src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2203,5 +2203,18 @@ private int getExactClasses(CORINFO_CLASS_STRUCT_* baseType, int maxExactClasses
             Debug.Assert(index <= maxExactClasses);
             return index;
         }
+
+#pragma warning disable CA1822 // Mark members as static
+        private CORINFO_CLASS_STRUCT_* getObjectType(void* objPtr)
+#pragma warning restore CA1822 // Mark members as static
+        {
+            // only strings are passed here for now
+            object obj = HandleToObject((IntPtr)objPtr);
+            if (obj is string)
```

Comment:
I do not think that this is correct. The object you will get for frozen strings is going to be `FrozenStringNode`. Also, you can get the type from `FrozenObjectNode` too.

Reply:
Changed, currently for NativeAOT only string objects can come here, I needed that API mostly for type objects (it helped to mitigate some regressions)

Reply:
I hope we will get https://github.com/dotnet/runtime/pull/76112#discussion_r979146309 as part of this PR so that native AOT is not falling behind. It means that we will want to handle non-string objects as well here.